docs/vector-library.md

```diff
@@ -0,0 +1,123 @@
+
+`vector.sign(vec: vector): vector`
+
+Applies `math.sign` component-wise for each vector. Equivalent of: `vector(math.sign(vec.x), ect)`.
+
+`vector.clamp(vec: vector, min: vector, max: vector): vector`
+
+Applies `math.clamp` component-wise for each vector. Equivalent of: `vector(math.clamp(vec.x, min.x, max.x), ect)`.
+
+`vector.max(...: vector): vector`
+
+Applies `math.max` component-wise for each vector. Equivalent of: `vector(math.max((...).x), ect)`.
+
+`vector.min(...: vector): vector`
+
+Applies `math.min` component-wise for each vector. Equivalent of: `vector(math.min((...).x), ect)`.
+
+---
+
+`vector.zero`
+
+Vector where `x=0, y=0, z=0, w?=0`.
+
+`vector.one`
+
+Vector where `x=1, y=1, z=1, w?=1`.
+
+---
+
+### Arithmetic operations
+
+Primitive operators for vectors are already implemented, so this RFC doesn't concern vector arithmetic.
+
+### Native codegen
+
+In the future, vectors will have special treatment in native codegen. This creates an expectation that the vector library will also have special treatment in codegen, but it isn't clear what that will look like, or if the performance benefits make sense.
+
+### Compiler options
+
+Currently, there are 2 compiler options relating to vectors, `vectorLib` & `vectorCtor`. This poses an interesting problem: a builtin library would remove the _requirement_ for such compiler options, however these options still need to be supported. The intuitive solution to this is to leave both compiler options working and maintained, but provide the built-in vector library by default, allowing two vector libraries and two vector constructors.
+
+### Typechecking
+
+A new primitive `vector` type should be added to the type system.
+
+## Drawbacks
+
+As per all additional globals, this creates a new global `vector`. This is a commonly used variable name, and many style guides will advise to avoid using `vector` as a variable name due to the global being added.
+
+Introducing a vector library means introducing more globals. Due to the prescence of vectors in performance-intensive code, it can be assumed that the built-in functions will have fastcalls. This means more fastcall slots will be used.
+
+Existing code won't get any improvements from a built-in library. Developers will have to switch over to the built-in library in order to gain any benefit.
+
+## Alternatives
+
+Do nothing; vectors have an internal constructor, which lets the runtime implement vectors, and subsequently a library for vector math.
+
+### Alternative implementations
+
+A more standard alternative to `vector(...)` could be something like `vector.create` or `vector.new`. This follows the standard set by `buffer.create`, `coroutine.create`, and `table.create`, and doesn't involve calling a table, which requires a metatable with `__call` to be set on the library. This breaks the standard previously set by Lua, as there is currently no library with a metatable set by default.
```

Comment:
string library sets the metatable of string values by default, so there is precedent (`luaopen_string`).

Reply:
To clarify, I mean there is no library which has an implicit `__call` metamethod on it. There's no luau library that lets you do, for example, `string()`.

Reply:
That's not what the RFC text says. It still says "there is currently no library with a metatable set by default" and that's not true. If it said that there's no library with a `__call` metamethod set on a type, that would be true.

Reply:
resolved